trin-core/src/portalnet/overlay.rs

```diff
@@ -343,14 +494,63 @@ impl OverlayProtocol {
     {
         let content_key = content_key.into();
         let msg = FindContent { content_key };
-        self.discovery
+        let result = self
+            .discovery
             .read_with_warn()
             .await
             .send_talkreq(
-                enr,
+                enr.clone(),
                 protocol,
                 Message::Request(Request::FindContent(msg)).to_bytes(),
             )
+            .await;
+        match result {
+            Ok(result) => {
+                self.process_response(result.clone(), enr, Default::default())
+                    .await
+                    .unwrap();
+                Ok(result)
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    // offer is sent in order to store content to k nodes with radii that contain content-id
+    // offer is also sent to nodes after FindContent (POKE)
+    pub async fn send_offer(
```

Comment:
Would be good to add `///` doc comment here. I know we don't do it for the similar `send_*` methods, but I think we should get in the habit of doing it for all public methods.